### PR TITLE
Send 'error' severity by default & document how to set it

### PR DESCRIPTION
**Pull request by kyrylo**
This is an upcoming feature.

### CODE REVIEW

File: src/Sharpbrake.Client/Model/Context.cs [hunk 1]
```diff
@@ -85,5 +85,11 @@ public class Context
         /// </summary>
         [JsonProperty("user")]
         public UserInfo User { get; set; }
+
+        /// <summary>
+        /// Error severity.
+        /// </summary>
+        [JsonProperty("severity")]
+        string Severity { get; set; }
```
Comment (aholovko) [inline src/Sharpbrake.Client/Model/Context.cs]: Plz add `public` modifier here.

Comment (kyrylo) [inline src/Sharpbrake.Client/Model/Context.cs]: Fixed.

**Comment by kyrylo** [comment]
Fixed build, added description to README.

PTAL

**Comment by aholovko** [comment]
> Fixed build, added description to README.

👍 

As stated in https://airbrake.io/docs/airbrake-faq/what-is-severity/ Airbrake supports well-defined list of severities (i.e. debug, info, notice, etc.). It makes sense to implement it as an `enum` that defines all supported values - it should help our customers to choose a proper value and avoid mistakes (that will be for sure in the case when arbitrary string values are allowed).

You can define it in the next way:

```csharp
public enum Severity
{
    Debug,
    Info,
    // etc...
}
```

I am not sure also whether it should be a part of `IHttpContext`. What if we have a console app that doesn't want to deal with HTTP. Shouldn't it have ability to set the severity of error as well?

**Comment by kyrylo** [comment]
👍 to enum

> I am not sure also whether it should be a part of IHttpContext. What if we have a console app that doesn't want to deal with HTTP. Shouldn't it have ability to set the severity of error as well?

Anything should be able to set severity. It's a basic feature and every app has its own understand of what's severe and what is not.

**Comment by aholovko** [comment]
> Anything should be able to set severity.

Yep, that's why I propose to set it not via HttpContext but using some other mechanism. One solution is to pass it via parameter to `NotifyAsync` method with default value set to `error`, e.g.

```csharp
NotifyAsync(Exception exception, IHttpContext context = null, Severity = Severity.Error)
```

If you don't mind, tomorrow we can work together on that part.

**Comment by kyrylo** [comment]
Why would we special case it? Currently, we pass `params` via `HttpContext`, but these params are not related to HTTP. They can be, however it's usually a place to store arbitrary info.

I don't see why Severity deserves special treatment. It's just another parameter in the context payload.

**Comment by aholovko** [comment]
>  Currently, we pass params via HttpContext, but these params are not related to HTTP.

From what I can see these parameters are HTTP related:
* [Params](https://github.com/airbrake/sharpbrake/blob/master/src/Sharpbrake.Client/Model/Notice.cs#L43) property from Notice
* `params` is _An object containing the request parameters. Where the key is the parameter name, e.g. { "page": "3", "sort": "desc" }._ from [spec](https://airbrake.io/docs/api/#create-notice-v3)
* implementation in [Middleware](https://github.com/airbrake/sharpbrake/blob/master/src/Sharpbrake.Http.Middleware/AspNetCoreHttpContext.cs#L52)
* implementation in [Module](https://github.com/airbrake/sharpbrake/blob/master/src/Sharpbrake.Http.Module/AspNetHttpContext.cs#L50)

**Comment by kyrylo** [comment]
I agree with this rationale. I didn't phrase it correctly. It's a place for HTTP params, yes. However, sometimes we need to store extra information and Params is the default place for that. Anything extra should go there.

```c#
NotifyAsync(Exception exception, IHttpContext context = null, Severity = Severity.Error)
```

^^ this code looks good, as long as we can support other fields (if we need them in the future). For example:

```c#
NotifyAsync(Exception exception, IHttpContext context = null, {
    Severity = Severity.Error,
    OtherThing = Val
})
```

This API looks acceptable. Would you be able to replace this PR with a different one? It'll take me quite some time to implement it.

**Comment by aholovko** [comment]
> However, sometimes we need to store extra information and Params is the default place for that. Anything extra should go there.

If this is a common place where severity is put (or will be put if this is a new feature) for other notifiers, then we should follow that convention as well. And I don't see any problems here. At the same time, we are not forced to use IHttpContext interface for that. We can use some other, more appropriate and less "semantic breaking" mechanism for passing the value of severity to `Param` collection. Using IHttpContext here also has another drawback - in console application you will be forced to have the full implementation of IHttpContext just to be able to set the value of severity. It seems to be quite inconvenient approach.

> this code looks good, as long as we can support other fields (if we need them in the future). For example...

Not sure that we can guarantee "as long as we can support other fields" part. I think that we should consider each case individually and make a decision based on the purpose of each particular field. If there will be a lot of such "custom" fields, perhaps we should come with some solution, but it's quite hard to talk about such solution in advance. Let's come back to error severity. As for me, this is a quite "important" field that deserves its own parameter in the `NotifyAsync` method. There are a lot of APIs where that kind of information has earned its own method and not just a parameter name (e.g. logger.Error(), logger.Warn(), etc.). I have updated your code to reflect that point. Please check these changes and let me know if you have any notes here.

**Comment by kyrylo** [comment]
Severity should be put in `context/severity`. 

Arbitrary params are useful for user filters. Let's say for some reason you want to send a custom environment variable. You define a new filter, read the value, now you need to attach it to the notice object. The params dict is the correct place for that. It's basically a dumpster for everything.

> Not sure that we can guarantee "as long as we can support other fields" part. I think that we should consider each case individually and make a decision based on the purpose of each particular field.

Sounds good.

> Please check these changes and let me know if you have any notes here.

Thanks! I have no objections. Hit merge as long as you are happy with the code. After that, we will want to make a new minor release.

**Pull request merged**